src/chainparams.cpp

```diff
@@ -38,6 +40,12 @@ static CBlock CreateGenesisBlock(const char* pszTimestamp, const CScript& genesi
     return genesis;
 }
 
+static CBlock CreateGenesisBlock(const char* pszTimestamp, const CScript& genesisOutputScript, uint32_t nTime, uint32_t nNonce, uint32_t nBits, int32_t nVersion, const CAmount& genesisReward)
```

Comment:
This function is only called from the next one and can be inlined. The change in the previous one is, I think all you need for your own CreateSignetGenesisBlock, which I would also prefer at the top (or at least before SigNetParams) in case it  turns into a static function in the future.
Really small nits.

Reply:
I don't think `CreateSignetGenesisBlock` will be static anytime soon, since it is also used in `mining.cpp` but I can definitely inline this one.

Reply:
Well, it can only be static if we stop supporting grinding the genesis block because the genesis block's pow is not checked anymore and we don't need it.
So I guess it depends on the other discussion. If we end up doing that, we won't need to move the function later. But it's a very minor nit, feel free to ignore.

Reply:
Good point. If we end up reversing the grinding approach I will try to remember to address this.

Reply:
One part about skipping genesis PoW that shouldn't be forgotten is that this doesn't just mean code changes in Bitcoin Core, but in every other software that checks the blockchain POW, e.g. btcd.

Reply:
well, they will need changes to support chains whose genesis block doesn't passs pow anyway, but yeah